### PR TITLE
metricbeat: Update beat module with apm-server monitoring metrics fields

### PR DESCRIPTION
**Pull request by carsonip**
## Proposed commit message

- Update metricbeat beat module mapping to contain latest apm-server monitoring metrics fields
- Update schema to parse `output` fields from stats
- Update tests

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## How to test this PR locally

- Start apm-server with monitoring
- Start standalone metricbeat with beat module
- Confirm that fields are mapped correctly in `metricbeat-*` data stream

Manual testing results described in https://github.com/elastic/apm-server/issues/13475

## Related issues

- Part of https://github.com/elastic/apm-server/issues/13475

### CODE REVIEW

File: metricbeat/module/beat/stats/data.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ var (
 		"cgroup":     c.Ifc("beat.cgroup"),
 		"system":     c.Ifc("system"),
 		"apm_server": c.Ifc("apm-server"),
+		"output":     c.Ifc("output"),
```
Comment (carsonip) [inline metricbeat/module/beat/stats/data.go:38]: [to reviewer] this is required for `beat.stats.output.elasticsearch.*`

File: metricbeat/module/beat/stats/data_test.go [hunk 2]
```diff
@@ -33,7 +33,7 @@ import (
 
 func TestEventMapping(t *testing.T) {
 
-	files, err := filepath.Glob("./_meta/test/stats.*.json")
+	files, err := filepath.Glob("./_meta/test/*.stats.*.json")
```
Comment (carsonip) [inline metricbeat/module/beat/stats/data_test.go:36]: [to reviewer] it wasn't testing apm-server.stats.712.json, not sure if it was intended.

Comment (klacabane) [reply metricbeat/module/beat/stats/data_test.go:36]: no idea but we'll know if CI fails :)

File: CHANGELOG.next.asciidoc [hunk 3]
```diff
@@ -175,6 +175,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Fix query logic for temp and non-temp tablespaces in Oracle module. {issue}38051[38051] {pull}39787[39787]
 - Set GCP metrics config period to the default (60s) when the value is below the minimum allowed period. {issue}30434[30434] {pull}40020[40020]
 - Fix missing metrics from CloudWatch when include_linked_accounts set to false. {issue}40071[40071] {pull}40135[40135]
+- Update beat module with apm-server monitoring metrics fields {pull}40127[40127]
```
[the remark this answers is not in the document]
Comment (klacabane) [reply CHANGELOG.next.asciidoc:178]: @elastic/beats would you know about that ?

**Review by klacabane — APPROVED** [review]
LGTM

**Pull request merged**